### ISSUE TITLE
question about your cross-validation & results reported in paper

### ISSUE THREAD

**Issue by kingofapplehead** [MCG-NJU/BCN#6]
Hi,

Thanks for sharing the code of your great work.

I noticed that you use n-fold cross validation for all 3 datasets. And I can see from your .bundle files that, in each split, you use 4/5 of all data for training and 1/5 of all data for testing (in the case of 5-fold cv). 

I am just wondering how you select the best model, for example, how do you know at which epoch to save the model as your best model to report testing results on the 1/5 test data? Do you further split the 4/5 training data into a train set and a val set and use the small val set for model selection? (Though I do not see this from the code.) Or, do you use the 1/5 test data both for selecting the best model and reporting the results?

Please correct me If I am very wrong in understanding your training process. 

Thanks!!

**Comment by zhenzhiwang** [comment]
Hi kingofapplehead,
The .bundle files are borrowed from the implementation of MS-TCN (https://github.com/yabufarha/ms-tcn). Following the evaluation code of MS-TCN, we choose the epoch which achieves the best average performance over all splits (e.g., 4 or 5), where we believe it is a standard practice in cross-validation. We do not further split the training data into training set and val set. In my understanding, the cross-validation itself is used to avoid the usage of val set if the total size of training data is small. Therefore, if someone need to train a final model to predict in the true test set, he should train the model in all splits and use the same epoch with the one achieving best average performance over all splits in the cross-validation stage. Sadly, the datasets of action segmentation task commonly do not have a real test set which can only be evaluated in the online test server, so researchers in this area commonly just report the cross-validation results.

Please feel free to comment here if you have further questions.

**Comment by kingofapplehead** [comment]
Hi, I see what you mean. Thanks for your kind explanations!

**Comment by VassMorozov** [comment]
Hi, I hope it is ok to comment on this closed issue, however my question is related to this. Happy to open a new issue if required. 

Are you reporting the average training accuracy for the 5 splits? If so, is there a reason that the average test accuracy was not shown as I would expect this to be the more meaningful result? I have also asked this question of the MS-TCN project, as after forking their repository, I have noted a significantly reduced average test accuracy. I would have thought that this may indicate overfitting but I am keen to get your views on this?

Thanks for the great resources and great work by the way!!

**Comment by zhenzhiwang** [comment]
> Are you reporting the average training accuracy for the 5 splits? If so, is there a reason that the average test accuracy was not shown as I would expect this to be the more meaningful result? 

Hi VassMorozov,

I do not understand your question well. First of all, all the reported accuracy results are average results from all splits (4 or 5) on the test set. In the cross-validation paradigm, the test set is a specific split and the training set is all other splits. If you mean the test accuracy differs from each split, I agree with you. But what is so called 'meaningful results'?

**Comment by VassMorozov** [comment]
Hi, thanks you have answered my question. I wasn't sure if average train or test was being reported! By meaningful results, I meant it makes more sense to see the test rather than the train as the train would of course be higher than the test.
Thanks again!
